keg_elements/sentry.py

```diff
@@ -68,9 +68,10 @@ def transform(self, data):
 
     def send(self, auth_header=None, **data):
         from keg import current_app
+
         if self.__log_reports__:
             # Report logging is enabled. Store the report in __report_log__
             self.__report_log__.append(data)
-        elif not current_app.config.get('TESTING'):  # pragma: no cover
+        elif not flask.has_app_context() or not current_app.config.get('TESTING'):  # pragma: no cover  # noqa
```

Comment:
Am I understanding this?

```python
has_app = flask.has_app_context()
if  (has_app and not current_app.config.get('TESTING')) or not has_app: 
    # Do report to the real SENTRY because we aren't under test
elif self.__log_reports__: 
   # do append
else:  # We 
   # RAISE something here the situation where we have an app and are not in testing and have not set `__log_reports__` is possible. For example, in a local dev environment?
```

Reply:
The alternative of the first condition you listed would be 
`has_app and (not has_app or current_app.config.get('TESTING'))` 
which can be simplified to
`has_app and current_app.config.get('TESTING')`
in which case we do not want to send the report

Reply:
But we also want to log if we _don't_ have the app, yes?